app/services/jwt_token.rb

```diff
@@ -36,6 +36,8 @@ def to_s
   # This method does not verify if the token signature is valid
   def decoded_payload
     @decoded_payload ||= JWT.decode(token, nil, false).first
+  rescue JWT::DecodeError => e
+    Foreman::Logging.exception('Failed to decode JWT', e)
```

Comment:
I don't think the service should be catching exceptions. Otherwise you can't distinguish between an invalid JSON content and `null`.